server/src/main/java/io/crate/execution/engine/collect/stats/JobsLogs.java

```diff
@@ -143,7 +144,7 @@ private void recordMetrics(JobContextLog log) {
      * <p>
      * {@link #logExecutionStart(UUID, String, Role, StatementClassifier.Classification)} is only called after a Plan has been created and execution starts.
      */
-    public void logPreExecutionFailure(UUID jobId, String stmt, String errorMessage, Role user) {
+    public void logPreExecutionFailure(UUID jobId, @NotNull String stmt, String errorMessage, Role user) {
```

Comment:
Note that we generally assume not-null for all parameters not explicitly marked as Nullable.
No need to add NotNull explicit. 

In this case it would even make it more confusing because it would input that jobId, errorMessage and user are nullable

Reply:
Thank you, removed them.